src/create.ts

```diff
@@ -82,7 +96,11 @@ export function create<
           return;
         }
 
-        wrappedSetDerivedState();
+        const newDerivedState = memoizedMapState(store.getState());
+
+        if (!shallowEqual(newDerivedState, lastStateRef.current)) {
+          setUpdates(updates => updates + 1);
```

Comment:
Add a comment that this causes the component to re-render using the new derived state (and that memozing mapState means this doesn't run mapState again).